recursion/recurse.rb

```diff
@@ -1,10 +1,32 @@
 def fact(n)
+  # base case: 
+  if n == 0 || n == 1
+    return 1
+  end
+    return n * fact(n - 1)
 end
 
 def fib(n)
+  # base cases:
+  if n == 0
+    return 0
+  elsif n == 1
+    return 1
+  end
+  return fib(n - 1) + fib(n - 2)
 end
 
 def pal(s)
+  s_array = s.split("")
+  #base case:
+  if s.length == 1 || s.length == 0
+    return true
+  end
+  if s_array[0] == s_array[-1]
+    s = pal(s_array.slice(1..-2).join("")) #weird syntax because of slice method
+  else
+    return false
```

Comment:
Would be nice to test for this first

return false if s[0] != s[-1]

because typically we test all base cases first, then recursive cases.